src/hooks/useRelatedVenues.tsx

```diff
@@ -10,15 +10,15 @@ import { WithId } from "utils/id";
 import { tracePromise } from "utils/performance";
 import { isTruthy } from "utils/types";
 
-import { useSovereignVenueId } from "./useSovereignVenueId";
+import { useSovereignVenue } from "./useSovereignVenue";
 
 const emptyArray: never[] = [];
 
 export interface RelatedVenuesContextState {
   isLoading: boolean;
   isError: boolean;
 
-  isSovereignVenueIdLoading: boolean;
+  isSovereignVenueLoading: boolean;
   sovereignVenue?: WithId<AnyVenue>;
   sovereignVenueId?: string;
```

Comment:
Please, also refactor RelatedVenues to not fetch `sovereignVenue` twice.

![image](https://user-images.githubusercontent.com/63194656/124479940-6cd75a00-ddaf-11eb-8bb3-e54f72b094d3.png)

You should pass sovereignVenue there

Reply:
@mike-lvov fixed in c0aa08769265d8bf04074054cae05bac8a0bd2db

Reply:
That depends on the logic of your hook actually.. because my code here will work regardless of the venueId provided. Whereas unless you changed the logic, you cache the very first sovereign venue ever used and never change it.

Reply:
Also, why would we change fetch related venues from being something generic that can be provided any venueId and work properly to requiring a full sovereign venue object to be passed into it? Seems like degraded functionality/not a great plan.

Reply:
> That depends on the logic of your hook actually.. because my code here will work regardless of the venueId provided. Whereas unless you changed the logic, you cache the very first sovereign venue ever used and never change it.
> 
> So changing this like that is probably a regression in functionality.

Disagree. If sovereignVenue is not working correctly, the app will be broken anyway. But with this change results in the performance gains by not fetching the sovereignVenue. Since you are using `await` there, it won't start fetching the descendants before fetching the sovereignVenue

> Also, why would we change fetch related venues from being something generic that can be provided any venueId and work properly to requiring a full sovereign venue object to be passed into it? Seems like degraded functionality/not a great plan.

It's only used in relatedVenues. I guess with sovereignVenue available, we can just use fetching the descendants though. 

> Whereas unless you changed the logic, you cache the very first sovereign venue ever used and never change it.

But you cache it too? You only fetch once given that it's in the context?

Reply:
> But you cache it too? You only fetch once given that it's in the context?

Then that should also be fixed.

Reply:
Given the above, my suggestion would be to revert the `fetchRelatedVenues` prev. functionality and to use `fetchDescendantVenues` in the `RelatedVenuesProvider`

Reply:
> my suggestion would be to revert to the fetchRelatedVenues prev. functionality

What does that mean?

Reply:
@0xdevalias 
![image](https://user-images.githubusercontent.com/63194656/124553648-a9519700-de3d-11eb-84c2-cf18d73648e9.png)

Reply:
Makes sense.

Reply:
So, you agree with using `fetchDescendantVenues` in the `RelatedVenuesProvider`?

Reply:
Seems fine to me, so long as it's properly combined and the array object from it memo'd/etc (not constructed inline in the function in a way that will change reference every time it's called)

Reply:
@0xdevalias IMO it'll be combined in the useAsync fn to prevent any reference changes

Reply:
Sounds good.

Reply:
Actually.. that risks re-running the `useAsync` and fetching everything all over again unless we're super careful.. I would probably suggest not including it there. Using deeply nested objects in react's deps is prime for issues unless everything is done perfectly.

Reply:
@0xdevalias @mike-lvov fixed in 98b0388

Reply:
@dmitrii-zolotukhin Wasn't @mike-lvov 's suggestion to reuse the sovereign venue from the hook, not to fetch it again here? As currently implemented in your recent change here it's the same problem as before, but made worse through more verbose/inefficient code.

Reply:
@0xdevalias @mike-lvov fixed in 2bcf61bbd342470f31bc588e68410eeb6a247708

Reply:
@dmitrii-zolotukhin nope, not it. I'll explain it to you in a call